Rename __USER_CAP* and *copy{in,out}cap*

brooksdavis opened the pull request:
- Rename `__USER_CAP*` and drop the `__` prefix following the model of PTRIN. (See #2485)
- Rename copy{in,out}cap variants to ptr
- Document copy{in,out}cap variants

brooksdavis committed:
ipf: copy{in,out}ptr -> copy{in,out}_indirect

These functions take the source or destintation address indirectly from
a potentially missaligned buffer.  Rename them to refect this and to
free up the copy{in,out}ptr names.

Some of the code in question is dead code and doesn't or won't compile,
but I've changed it all for consistency.

NB: If the pointers are actually stored under aligned then this code is
broken with CHERI.

brooksdavis committed:
sys: __USER_CAP -> USER_PTR

Rename the __USER_CAP macros to USER_PTR.  This reflects our
desire to talk about pointers when ever possible rather than
capabilities.

brooksdavis committed:
sys: copy{in,out}cap -> copy{in,out}ptr

Rename all variants of copy{in,out}cap to copy{in,out}ptr.  This
reflects our desire to talk about pointers when ever possible rather
than capabilities.

brooksdavis committed:
copy(9): document copy{in,out}ptr*

brooksdavis committed:
socket(9): document sooptcopy{in,out}ptr

brooksdavis committed:
USER_PTR_STR -> USER_PTR_PATH

Migrate many instances of USER_PTR_STR to USER_PTR_PATH to reduce the
number of unbounded capabilities created.

brooksdavis committed:
freebsd64: remove duplicate struct jail64 definition

brooksdavis committed:
sys/abi_compat.h: relocate USER_*PTR* from systm.h

These macros relate to compatibility with integer-pointer ABIs so put
them with other compat macros rather than cluttering sys/systm.h with
symbols most consumers don't need.

brooksdavis committed:
sys: __USER_CODE_CAP -> USER_CODE_PTR

This reflects our desire to talk about pointers when ever possible
rather than capabilities.

bsdjhb reviewed (commented):
Generally looks good to me

brooksdavis committed:
drm_atomic_uapi.c: include sys/abi_compat.h unconditionally

A control interface isn't passing a pointer to userspace properly
leading to the need to create a pointer from an address.  This is
a bug and potentially a write-what-where gadget.

brooksdavis commented on sys/sys/systm.h:
```diff
@@ -140,7 +140,7 @@ extern bool scheduler_stopped;
 
-#define	__USER_CODE_CAP(ptr)						\
+#define	USER_CODE_PTR(ptr)						\
 	___USER_CFROMPTR((ptr), __USER_PCC, __USER_PCC_OFFSET_ENABLED)
```
Patches in #2503
